mayastor/src/grpc/mod.rs

```diff
@@ -33,6 +33,24 @@ mod mayastor_grpc;
 mod nexus_grpc;
 mod server;
 
+use crate::{bdev::VerboseError, subsys};
 pub use server::MayastorGrpcServer;
 use tonic::{Response, Status};
+
 pub type GrpcResult<T> = std::result::Result<Response<T>, Status>;
+
+/// exports the current configuration to the optional config file
+pub fn export_config() -> std::result::Result<(), Status> {
```

Comment:
What about, if we were to put this function within the config subsystem itself and create a function that accepts a closure that writes on success

```rust
async fn sync<F, T, E>( future: F) -> Result<T, E>
where
    F: Future<Output = Result<T, E>>,
{
    let cfg = ....;
    let result = future.await;
    if result.is_ok() {
        cfg.write();
    }
    result
}
```

Then we can use it like:
```rust
async fn grpc_thing(&self, request: Request<Null>) -> GrpcResult<Null> {
  Config::sync(  async { /* body that was there already */  } ).await
}
```

This way we can easily see which functions require/want to save?

Reply:
yeah that's much better :+1: